### PR TITLE
Bump CMake version to 3.19.6.

### PR DESCRIPTION
**Pull request by zoecarver**
Updates the required CMake version for Swift and the Swift Benchmarks to ~~3.18~~ 3.19.6. Updates the docs to reflect this change. Doesn't modify the required version for building the stdlib.

Required for #36082.

### CODE REVIEW

**Comment by zoecarver** [comment]
@shahmishal This patch would require some of the Linux build bots to be updated.

**Comment by gottesmm** [comment]
Can you do a post on the forum about this to check for objections?

**Comment by buttaface** [comment]
I think you'll need to update the CMake built from source too, see #30695 for an example.

**Comment by zoecarver** [comment]
@buttaface will do. 


---
[Here's the forum post.](https://forums.swift.org/t/bump-cmake-version-to-3-18/45033)

**Comment by buttaface** [comment]
Anything holding this back? There is [a Foundation patch that I'd like to upstream](https://github.com/buttaface/termux-packages/blob/swift54/packages/swift/foundation-rpath.patch) that needs a CMake bump first, as the last `BUILD_RPATH` line from my patch doesn't work with the currently used year-old CMake.

**Comment by zoecarver** [comment]
There's nothing on my side holding this back. I think @gottesmm was waiting on something, though. I also might just try to get uswift to use spm to unblock #36082.

**Comment by zoecarver** [comment]
@gottesmm any updates or should I try to get swiftpm working?

**Comment by zoecarver** [comment]
@swift-ci please test.

**Comment by zoecarver** [comment]
@swift-ci please test.

**Comment by gottesmm** [comment]
@zoecarver can you add to the commit message that you are updating the cmake tag for 3.19 in update-checkout?

**Comment by artemcm** [comment]
@swift-ci please test

**Comment by zoecarver** [comment]
@gottesmm done. 

@swift-ci please smoke test.

File: docs/HowToGuides/GettingStarted.md
Comment (buttaface) [inline docs/HowToGuides/GettingStarted.md:190]: I'd like to point out something here. Based on my reading of [the CMake docs](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html) and usage over the years, the real de facto minimum after this pull will be 3.19.1, as that's what the CI will be enforcing. In that case, that's what should be listed as the minimum here in the docs.

@compnerd, can you confirm?

Comment (gottesmm) [reply docs/HowToGuides/GettingStarted.md:190]: One thing to be aware of is that the cmake issue you are talking about would only apply on Linux where the build behavior occurs. Windows/macOS on CI will just use what is on the system.

**Comment by zoecarver** [comment]
Looks like there was a version mismatch issue. Maybe a clean build will get the bots green. 

@swift-ci please clean test.

File: docs/HowToGuides/GettingStarted.md
Comment (buttaface) [reply docs/HowToGuides/GettingStarted.md:190]: Sure, the de facto minimum is the minimum CMake version used on the CI, ie `min(mac CI version, linux CI version, Win CI version)`. Raising this for the Linux CI will simply change one input to that equation.

**Comment by buttaface** [comment]
Looks like this is moving forward in #37517, maybe this pull can just add the doc changes after that's in.

**Comment by zoecarver** [comment]
@gottesmm do you want me to update this to 3.19? It looks like the other PR only updated the checkouts script.

**Comment by gottesmm** [comment]
@zoecarver not time yet. Waiting for one other thing.

**Comment by gottesmm** [comment]
and I am going to want to update to 3.19.6

**Comment by zoecarver** [comment]
@swift-ci please clean test.

**Comment by zoecarver** [comment]
@swift-ci please clean test macOS.

**Comment by zoecarver** [comment]
@swift-ci please test macOS.

**Comment by zoecarver** [comment]
@swift-ci please test Windows.

**Comment by zoecarver** [comment]
@swift-ci please test Windows.

**Review by shahmishal — APPROVED** [review]

**Comment by zoecarver** [comment]
Source kit issues :/

@swift-ci please test Windows.

**Comment by zoecarver** [comment]
@swift-ci please test Windows.

**Comment by gottesmm** [comment]
FYI: Even when this is green, please don't merge until Sat or Sun. There is a bunch of commits trying to get into main right now and we don't want to churn the tree with something unless we need to.

**Comment by zoecarver** [comment]
The windows bots keep failing for unrelated issues. This shows that there isn't a problem with the CMake version, though, because those tests wouldn't even be able to run if there was. I'll give this one more try because it would be nice if it was green, but otherwise I'll just land it tomorrow. 

@swift-ci please test Windows.

**Comment by compnerd** [comment]
CC: @drodriguez - this seems to have regressed the VS2017 builders :-(.

**Comment by drodriguez** [comment]
> CC: @drodriguez - this seems to have regressed the VS2017 builders :-(.

Hopefully the next build will use CMake 3.21. Thanks for the heads up!

**Pull request merged**
